src/OpenConext/EngineBlockApiClientBundle/Service/AttributeReleasePolicyService.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/**
+ * Copyright 2016 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace OpenConext\EngineBlockApiClientBundle\Service;
+
+use OpenConext\EngineBlockApiClientBundle\Http\JsonApiClient;
+use OpenConext\Profile\Value\Consent;
+use OpenConext\Profile\Value\ConsentList;
+use OpenConext\Profile\Value\SpecifiedConsent;
+use OpenConext\Profile\Value\SpecifiedConsentList;
+use OpenConext\ProfileBundle\Attribute\AttributeSetWithFallbacks;
+use Surfnet\SamlBundle\SAML2\Attribute\Attribute;
+use Surfnet\SamlBundle\SAML2\Attribute\AttributeSetInterface;
+
+final class AttributeReleasePolicyService
+{
+    /**
+     * @var JsonApiClient
+     */
+    private $jsonApiClient;
+
+    public function __construct(JsonApiClient $jsonApiClient)
+    {
+        $this->jsonApiClient = $jsonApiClient;
+    }
+
+    /**
+     * @param ConsentList $consentList
+     * @param AttributeSetInterface $attributeSet
+     * @return SpecifiedConsentList
+     */
+    public function applyAttributeReleasePolicies(ConsentList $consentList, AttributeSetInterface $attributeSet)
```

Comment:
I think this method is kinda messy. This is because we have a ConsentList and an AttributeSet and we get an array from the response that turns into a SpecifiedConsentList. We might want to find a way to separate the mapping logic from the other service logic.